catalog/app/containers/Bucket/CopyButton.js

```diff
@@ -0,0 +1,145 @@
+import * as React from 'react'
+import * as M from '@material-ui/core'
+import * as Lab from '@material-ui/lab'
+
+import * as AWS from 'utils/AWS'
+import AsyncResult from 'utils/AsyncResult'
+import { useData } from 'utils/Data'
+
+import * as requests from './requests'
+
+function MenuPlaceholder() {
+  const t = M.useTheme()
+
+  return (
+    <M.Box minWidth={t.spacing(22)}>
+      <M.MenuItem disabled>
+        <Lab.Skeleton height={t.spacing(6)} width="100%" />
+      </M.MenuItem>
+      <M.MenuItem disabled>
+        <Lab.Skeleton height={t.spacing(6)} width="100%" />
+      </M.MenuItem>
+      <M.MenuItem disabled>
+        <Lab.Skeleton height={t.spacing(6)} width="100%" />
+      </M.MenuItem>
+    </M.Box>
+  )
+}
+
+function MenuItem({ item, onClick }) {
+  return (
+    <M.MenuItem onClick={React.useCallback(() => onClick(item), [item, onClick])}>
+      <M.ListItemText primary={item.name} secondary={item.url} />
+    </M.MenuItem>
+  )
+}
+
+// TODO: use ref according to proper use of M.Menu
+const BucketsListFetcher = React.forwardRef(function BucketsListFetcher(
+  { bucket, children },
+  ref,
+) {
+  const s3 = AWS.S3.use()
+  const data = useData(requests.workflowsList, { s3, bucket })
+  return children(data.result, ref)
+})
+
+function SuccessorsSelect({ anchorEl, bucket, open, onChange, onClose }) {
+  // FIXME: add documentation link
+  return (
+    <M.Menu anchorEl={anchorEl} onClose={onClose} open={open}>
+      <BucketsListFetcher bucket={bucket}>
```

Comment:
to avoid complexities with ref forwarding etc i think we should use a hook for fetching the data here -- move the useData call from BucketsListFetcher to this component -- this will also achieve successors caching

Reply:
Still need to use `<div>` as a wrapper for `M.MenuItem`s to avoid `ref` issue

Reply:
i tried it locally and it works. what's exactly the issue you're having?

Reply:
Console's error for Menu `ref`

I fixed it by forwarding `ref` for `MenuItem`. I removed `div`, by the way

Reply:
it's strange, bc i've tried it locally (edited the code) and there were no errors. well, nvm